Lib/fontTools/otlLib/builder.py

```diff
@@ -328,12 +328,42 @@ def build(self):
         rulesets = self.rulesets()
         chaining = any(ruleset.hasPrefixOrSuffix for ruleset in rulesets)
         for ruleset in rulesets:
-            for rule in ruleset.rules:
-                subtables.append(self.buildFormat3Subtable(rule, chaining))
+            if not ruleset.hasAnyGlyphClasses and len(ruleset.rules) > 2:
+                # Experimentally, format 1 is only more efficient for >2 rules
+                subtables.append(self.buildFormat1Subtable(ruleset, chaining))
+            else:
+                for rule in ruleset.rules:
+                    subtables.append(self.buildFormat3Subtable(rule, chaining))
         # If we are not chaining, lookup type will be automatically fixed by
         # buildLookup_
         return self.buildLookup_(subtables)
 
+    def buildFormat1Subtable(self, ruleset, chaining=True):
+        st = self.newSubtable_(chaining=chaining)
+        st.Format = 1
+        rulesetObjs = OrderedDict()
+        for rule in ruleset.rules:
+            (prefix, glyphs, suffix, lookups) = rule
+            firstGlyph = list(glyphs[0])[0] # odict_keys can't be indexed, needs coercion
```

Comment:
the fact odict_keys can't be indexed is obvious -- what is not is whether the `rule` tuple should contain odict_keys as items. Was that an explicit/documented decision? For consistency it would be better that the type of this ruleset rules be  and always the same (be it list, tuple, or whatever). So this coercion to list could be done when the rule tuple is first created, instead of down the line.

Besides, these rules ideally could be turned into a named tuple and their items accessed with attribute lookup (e.g. `rule.prefix`) instead of unpacking the tuple to get to its items by position.